shared/src/ledger/tx.rs

```diff
@@ -1337,72 +1337,79 @@ pub async fn build_pgf_stewards_proposal<
 /// Submit an IBC transfer
 pub async fn build_ibc_transfer<C: crate::ledger::queries::Client + Sync>(
     client: &C,
-    args::TxIbcTransfer {
-        tx: tx_args,
-        source,
-        receiver,
-        token,
-        amount,
-        port_id,
-        channel_id,
-        timeout_height,
-        timeout_sec_offset,
-        memo,
-        tx_code_path,
-    }: args::TxIbcTransfer,
+    mut args: args::TxIbcTransfer,
     gas_payer: &common::PublicKey,
 ) -> Result<Tx, Error> {
     // Check that the source address exists on chain
     let source =
-        source_exists_or_err(source.clone(), tx_args.force, client).await?;
+        source_exists_or_err(args.source.clone(), args.tx.force, client)
+            .await?;
     // We cannot check the receiver
 
+    // validate the amount given
+    let validated_amount =
+        validate_amount(client, args.amount, &args.token, args.tx.force)
+            .await
+            .expect("expected to validate amount");
+    if validated_amount.canonical().denom.0 != 0 {
+        return Err(Error::Other(format!(
+            "The amount for the IBC transfer should be an integer: {}",
+            validated_amount
+        )));
+    }
+    let validate_fee = validate_amount(
+        client,
+        args.tx.gas_amount,
+        &args.tx.gas_token,
+        args.tx.force,
+    )
+    .await
+    .expect("expected to be able to validate fee");
+
+    args.amount = InputAmount::Validated(validated_amount);
+    args.tx.gas_amount = InputAmount::Validated(validate_fee);
```

Comment:
I feel like this is unnecessary. can't you use the inner amounts directly?

Reply:
You're right. For IBC transfers, we don't have to update them.